docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,288 @@ keywords:
+- **an extension point ID** _(string)_
+  - In case it's an extension point in core Grafana, it must start with `grafana/`
+  - In case it's inside a plugin, it must start with `plugin/<PLUGIN_ID>/`
+  - It must be unique
+- **an app plugin** - _currently only app plugins can create extension points_
 
-## Create an extension point
+:::note
+
+When designing the UI make sure the extension point supports a scenario where multiple extensions can be added without breaking the UI. Also consider if there is any information from the current view that should be shared with the extensions added to the extension point. It could be information from the current view that could let the extending plugin pre-fill values or other data in the extension's functionality.
 
-Use the `getPluginLinkExtensions` method in `@grafana/runtime` to create an extension point within your plugin.
+:::
+
+## Create an extension point
 
 :::danger
 
 When you create an extension point in a plugin, you create a public interface for other plugins to interact with. Changes to the extension point ID or its context could break any plugin that attempts to register a link inside your plugin.
 
 :::
 
-The `getPluginLinkExtensions` method takes an object consisting of the `extensionPointId`, which must begin `plugins/<PLUGIN_ID>`, and any contextual information that you want to provide. The `getPluginLinkExtensions` method returns a list of extension links that your program can then loop over.
+You can easily create an extension point using the following functions (they live in `@grafana/runtime`) to fetch extensions for a certain extension point ID:
+
+### The `usePluginExtensions()` hook
+
+:::note
+
+In case the `context` object is created dynamically, make sure to wrap it into a `useMemo()` to prevent unnecesssary rerenders. [More info](#)
+
+:::
+
+The `usePluginExtensions` React hooks returns the list of extensions that are registered for a certain extension point ID. The hook dynamically updates its return value when the list of extensions changes. This behavior usually happens when extensions are registered during runtime due to dynamic plugin loading.
+
+#### Syntax
+
+```tsx
+usePluginExtensions(options);
+usePluginLinkExtensions(options); // Only returns extensions that have type `type="link"`
+usePluginComponentExtensions(options); // Only returns extensions that have type `type="component"`
+```
+
+#### Parameters
+
+##### `options.extensionPointId` - _string_
+
+The unique identifier of your extension point. It must begin with `plugins/<PLUGIN_ID>`. For example: `plugins/myorg-super-app`.
+
+##### `options?.context` - _object (Optional)_
+
+An object containing information related to your extension point that you would like to share with the extensions. For example: `{ baseUrl: '/foo/bar' }`. This parameter is not available for component extensions, instead you should pass contextual information using the component props.
```

Comment:
> instead you should pass contextual information using the component props

We make this statement a lot... so a few questions:
- should we be providing advice to duplicate contextual information in both manners? If that is the case, why is it on the extension point developer? My worry is that it is up to the _extension_ developer to decide whether theres is a link or component, but by extending the same extension point they may get different context available, feels confusing.
- We reference it in the `Example - rendering component extensions` but maybe it is worth having an even smaller snippet which we could then link to from any time we make this statement?

Reply:
> We reference it in the Example - rendering component extensions but maybe it is worth having an even smaller snippet which we could then link to from any time we make this statement?

Yeah, I think that's a good idea! How about creating that sub-section "Considerations..." or "Best practices" and adding this as a point under there?


> should we be providing advice to duplicate contextual information in both manners? If that is the case, why is it on the extension point developer? My worry is that it is up to the extension developer to decide whether theres is a link or component, but by extending the same extension point they may get different context available, feels confusing.

Hmm, I don't think so, the two context objects should be the same, the difference is in how the extensions receive it. Probably the communication of this is not clear though, I think we should have a chat about it and try to make sure it's easier to understand.